cmd/kaspawallet/libkaspawallet/bip32/extended_key.go

```diff
@@ -0,0 +1,98 @@
+package bip32
+
+import (
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/kaspanet/go-secp256k1"
+	"github.com/pkg/errors"
+)
+
+type ExtendedKey struct {
```

Comment:
The first version I wrote used two types with a lot of code duplication, and then I had a bug specific to the public key child derivation. Because of the massive amount of code duplication I felt like it'll be less error prone to unite them.

Reply:
I don't mind that you unite them implementation-wise, but then expose to the outside world as separate beings.

Reply:
But if you parse a key it can give you both of them. So should it return an interface?

Reply:
Hmmm... oh well I guess.